From patchwork Mon Oct 15 20:46:13 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: "Tambe, William" <WILLIAM.TAMBE@amd.com>
X-Patchwork-Id: 10642461
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 9C0771057
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Oct 2018 20:46:22 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 8B75928AFF
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Oct 2018 20:46:22 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 7E39229053; Mon, 15 Oct 2018 20:46:22 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 74B6728AFF
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Oct 2018 20:46:20 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726068AbeJPEdM (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 16 Oct 2018 00:33:12 -0400
Received: from mail-sn1nam02on0065.outbound.protection.outlook.com
 ([104.47.36.65]:7320
        "EHLO NAM02-SN1-obe.outbound.protection.outlook.com"
        rhost-flags-OK-OK-OK-FAIL) by vger.kernel.org with ESMTP
        id S1725924AbeJPEdM (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 16 Oct 2018 00:33:12 -0400
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=amdcloud.onmicrosoft.com; s=selector1-amd-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=6QLcVh4rMTCAre4+tJj6C4KnbTDdavj//KHzJB+tHro=;
 b=mgLwBka1ZFTvUE9dZrrsyzzpgV5rTwO2W6gNyhYsVuyNUlqPXjxlIhIqtIjqMDf4HBqXZGOYY0LG0eqyDatC5AkWm6PW2Zcvn47slm67GWtuTVfc5ZEhUyq+WWw7AuoTGMv5Uwj1Vab5YiXQsSdhGSs1pkH5UURuhpXqPH36AYc=
Received: from DM5PR1201MB0091.namprd12.prod.outlook.com (10.174.106.21) by
 DM5PR1201MB0187.namprd12.prod.outlook.com (10.174.107.11) with Microsoft SMTP
 Server (version=TLS1_2, cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id
 15.20.1228.26; Mon, 15 Oct 2018 20:46:13 +0000
Received: from DM5PR1201MB0091.namprd12.prod.outlook.com
 ([fe80::5084:cdf4:32fb:977e]) by DM5PR1201MB0091.namprd12.prod.outlook.com
 ([fe80::5084:cdf4:32fb:977e%4]) with mapi id 15.20.1228.020; Mon, 15 Oct 2018
 20:46:13 +0000
From: "Tambe, William" <William.Tambe@amd.com>
To: "kvm@vger.kernel.org" <kvm@vger.kernel.org>
Subject: [kvm-unit-tests PATCH] x86: svm: Fix incorrect ordering within struct
 regs
Thread-Topic: [kvm-unit-tests PATCH] x86: svm: Fix incorrect ordering within
 struct regs
Thread-Index: AQHUZMgchRsr50Ik7kqKgTnopLr03Q==
Date: Mon, 15 Oct 2018 20:46:13 +0000
Message-ID: <20181015204553.18699-1-william.tambe@amd.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-clientproxiedby: SN4PR0501CA0047.namprd05.prod.outlook.com
 (2603:10b6:803:41::24) To DM5PR1201MB0091.namprd12.prod.outlook.com
 (2603:10b6:4:54::21)
authentication-results: spf=none (sender IP is )
 smtp.mailfrom=William.Tambe@amd.com;
x-ms-exchange-messagesentrepresentingtype: 1
x-originating-ip: [165.204.77.1]
x-ms-publictraffictype: Email
x-microsoft-exchange-diagnostics: 
 1;DM5PR1201MB0187;20:0VWO8tWzXc6DEcWFHxYdlw/fcF0BfchMD7c9+OmGYY6PrN5K503tU6lin/ZBpEVAhUuUETlTKduZhAoX7r+XSQ677etMJpOYCUw/W7ZrCxEch3y1rX5EeliUwD32ddKkXF4WVKpzGYO72vFGsREHHaWPp2AxXk/zEwCx/8ql4k6l2Lp4ZxcukEhRO9YRwqXE64MwJ/tCvsblaCCGl3dIr4dQmlc4zA1rOtjsum5iDM/eJ5JJ2IVt1e2Nr21fnu02
x-ms-office365-filtering-correlation-id: 0ab7e08a-ddc0-4bc7-e30e-08d632df3e70
x-ms-office365-filtering-ht: Tenant
x-microsoft-antispam: 
 BCL:0;PCL:0;RULEID:(7020095)(4652040)(8989299)(5600074)(711020)(4618075)(4534185)(4627221)(201703031133081)(201702281549075)(8990200)(2017052603328)(7153060)(7193020);SRVR:DM5PR1201MB0187;
x-ms-traffictypediagnostic: DM5PR1201MB0187:
x-microsoft-antispam-prvs: 
 <DM5PR1201MB01871FB7763E897C81316E79F7FD0@DM5PR1201MB0187.namprd12.prod.outlook.com>
x-exchange-antispam-report-test: UriScan:(767451399110);
x-ms-exchange-senderadcheck: 1
x-exchange-antispam-report-cfa-test: 
 BCL:0;PCL:0;RULEID:(8211001083)(6040522)(2401047)(5005006)(8121501046)(823302103)(3231355)(944501410)(52105095)(3002001)(10201501046)(93006095)(93001095)(6055026)(149066)(150057)(6041310)(201703131423095)(201702281528075)(20161123555045)(201703061421075)(201703061406153)(20161123560045)(20161123564045)(20161123562045)(20161123558120)(201708071742011)(7699051);SRVR:DM5PR1201MB0187;BCL:0;PCL:0;RULEID:;SRVR:DM5PR1201MB0187;
x-forefront-prvs: 0826B2F01B
x-forefront-antispam-report: 
 SFV:NSPM;SFS:(10009020)(346002)(39860400002)(396003)(376002)(366004)(136003)(199004)(189003)(478600001)(72206003)(316002)(5250100002)(256004)(14444005)(2501003)(6916009)(2616005)(305945005)(5660300001)(7736002)(476003)(102836004)(186003)(14454004)(36756003)(26005)(486006)(8676002)(1730700003)(6116002)(6436002)(2351001)(3846002)(52116002)(66066001)(8936002)(81166006)(71200400001)(71190400001)(86362001)(99286004)(81156014)(97736004)(2900100001)(25786009)(6506007)(386003)(105586002)(53936002)(106356001)(6512007)(2906002)(6486002)(1076002)(68736007)(5640700003);DIR:OUT;SFP:1101;SCL:1;SRVR:DM5PR1201MB0187;H:DM5PR1201MB0091.namprd12.prod.outlook.com;FPR:;SPF:None;LANG:en;PTR:InfoNoRecords;A:1;MX:1;
received-spf: None (protection.outlook.com: amd.com does not designate
 permitted sender hosts)
x-microsoft-antispam-message-info: 
 T+BZUcgIZW6MArip32WDyCzw8YPYKQhYO+WxGRMXK4iKT5fvVecuu6v5borevZ7ef2ZXBmwrFW444zEx2KyjBm8wrcJ+tVginVSErcCoQCLIOxZD8WjLjn0SMqqi0S+a6QvZiXSfrMTxiBaAwmfaeEgk+gzGRaJRChnb/b7Duw/g298KUyc1WbmqxH+qo0vtJ4TP2pgFl0ZBPuxXG8COxL6SfMcJd2I9V59d/5VmwqrXs7EFtmeweaDpc73134ZIERqhNn/SrcgoFOT1gAypF2BxPGvZkaQwa+so9tOdGQ+OlZEDeLj2sE7RYQu116ayAJUd0H4LI6HWUmTSNxGuIsbSvdVpM2f1J1qseHMyW9w=
spamdiagnosticoutput: 1:99
spamdiagnosticmetadata: NSPM
Content-Type: text/plain; charset="iso-8859-1"
MIME-Version: 1.0
X-OriginatorOrg: amd.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 0ab7e08a-ddc0-4bc7-e30e-08d632df3e70
X-MS-Exchange-CrossTenant-originalarrivaltime: 15 Oct 2018 20:46:13.1653
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 3dd8961f-e488-4e60-8e11-a82d994e183d
X-MS-Exchange-Transport-CrossTenantHeadersStamped: DM5PR1201MB0187
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

"struct regs" is used within x86/svm.c to describe
the memory region where SAVE_GPR_C saves registers.

The ordering within "struct regs" must be corrected
so that it matches how SAVE_GPR_C stores registers.

Signed-off-by: William Tambe <william.tambe@amd.com>
---
 x86/svm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/svm.c b/x86/svm.c
index 701ab2f..bc0396d 100644
--- a/x86/svm.c
+++ b/x86/svm.c
@@ -199,9 +199,9 @@ static void test_thunk(struct test *test)
 
 struct regs {
         u64 rax;
+        u64 rbx;
         u64 rcx;
         u64 rdx;
-        u64 rbx;
         u64 cr2;
         u64 rbp;
         u64 rsi;
